### ISSUE TITLE
Chart corrupted after update

### ISSUE THREAD

**Issue by zdm** [amcharts/amcharts5#1126]
Bug was introduced in the version: `5.4.9`.

Please, click `Apply dataset 1`, then `Apply dataset 2`, then `Apply dataset 1` again.
You will see, that data is not displayed.

```html
<style>
    #chartdiv {
        width: 100%;
        height: 500px;
    }
</style>

<script src="https://cdn.amcharts.com/lib/5/index.js"></script>
<script src="https://cdn.amcharts.com/lib/5/xy.js"></script>
<script src="https://cdn.amcharts.com/lib/5/themes/Animated.js"></script>

<script>
    var series1;

    am5.ready(function () {
        var root = am5.Root.new("chartdiv");

        root.setThemes([am5themes_Animated.new(root)]);

        const chart = root.container.children.push(
            am5xy.XYChart.new(root, {
                layout: root.verticalLayout,
                panX: true,

                // "wheelX": "panX",
                pinchZoomX: true,

                // "panY": true,
                // "wheelY": "zoomX",
            })
        );

        // x axis
        const xAxis = chart.xAxes.push(
            am5xy.DateAxis.new(root, {
                baseInterval: {
                    timeUnit: "day",
                    count: 1,
                },
                renderer: am5xy.AxisRendererX.new(root, {}),
                tooltipDateFormat: "i",
                tooltip: am5.Tooltip.new(root, {}),
            })
        );

        // y axis
        const yAxis = chart.yAxes.push(
            am5xy.ValueAxis.new(root, {
                renderer: am5xy.AxisRendererY.new(root, {}),
                tooltip: am5.Tooltip.new(root, {}),
            })
        );

        // series 1
        series1 = chart.series.push(
            am5xy.SmoothedXLineSeries.new(root, {
                name: "series 1",
                xAxis,
                yAxis,
                valueXField: "date",
                valueYField: "value1",
                tooltip: am5.Tooltip.new(root, {
                    labelText: "{valueY}",
                }),
                stroke: am5.color("#00ff00"),
                fill: am5.color("#00ff00"),
                connect: true, // XXX
            })
        );

        // fill settings
        series1.fills.template.setAll({
            fillOpacity: 0.5,
            visible: true,
        });

        // date processor
        series1.data.processor = am5.DataProcessor.new(root, {
            dateFields: ["date"],
            dateFormat: "i",
        });

        // scroll bar
        chart.set(
            "scrollbarX",
            am5.Scrollbar.new(root, {
                orientation: "horizontal",
            })
        );

        // cursor
        chart.set(
            "cursor",
            am5xy.XYCursor.new(root, {
                behavior: "zoomX",
            })
        );

        // legend
        const legend = chart.children.push(
            am5.Legend.new(root, {
                centerX: am5.p50,
                x: am5.p50,
            })
        );

        legend.data.setAll(chart.series.values);
    });

    function dataset1() {
        const data = [
            {
                date: "2023-01-01",
                value1: 1,
            },
            {
                date: "2023-01-02",
                value1: 2,
            },
            {
                date: "2023-01-03",
                value1: 3,
            },
        ];

        series1.data.setAll(data);
    }

    function dataset2() {
        const data = [
            {
                date: "2023-01-01",
                value1: 0,
            },
            {
                date: "2023-01-02",
                value1: 0,
            },
            {
                date: "2023-01-03",
                value1: 0,
            },
        ];

        series1.data.setAll(data);
    }
</script>

<!-- HTML -->
<div id="chartdiv"></div>

<button type="button" onclick="dataset1()">Apply dataset 1</button>
<br />
<button type="button" onclick="dataset2()">Apply dataset 2</button>

```

**Comment by zdm** [comment]
fixed in `5.4.11`.

**Comment by martynasma** [comment]
Fixed in [5.4.11](https://github.com/amcharts/amcharts5/releases/tag/5.4.11).

## [5.4.11] - 2023-10-10

### Fixed
- `SliceGrouper` would use first color in series for a "group" slice if series was using a color set with custom list of colors.
- Fixed a problem where no series was displayed after data with all 0 values was set (since `5.4.10`).
- `forceInactive` was not working on `Container`.
- `drawingsInteractive(value: bool)` method in `StockChart` was not working properly.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
